### ISSUE TITLE
Lint/LiteralInInterpolation has incorrect autocorrects for embedded quotes and escaped interpolation text

### ISSUE THREAD

**Issue by RedZapdos123** [rubocop/rubocop#15250]
## Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

## Issue type

Incorrect autocorrect

## Cop name

`Lint/LiteralInInterpolation`

## Steps to reproduce the problem

I reproduced this on the latest stable RuboCop release and also on current `master` at `21f0cb9`.

### Repro 1: invalid syntax after autocorrect

1. Create a file `example.rb` with the following content:

```ruby
puts "x#{"\""}y"
```

2. Run:

```bash
bundle exec ruby exe/rubocop example.rb --only Lint/LiteralInInterpolation -A
ruby -c example.rb
```

### Repro 2: behavior change after autocorrect

1. Create a file `example.rb` with the following content:

```ruby
puts "x#{"\\#{1}"}y"
```

2. Run:

```bash
bundle exec ruby exe/rubocop example.rb --only Lint/LiteralInInterpolation -A
ruby example.rb
```

## Expected behavior

Autocorrect should either preserve valid Ruby syntax and runtime behavior, or skip autocorrection if that cannot be done safely.

For these examples:

- the first example should remain valid Ruby after autocorrect
- the second example should keep printing `x#{1}y`

### Repro 1

Autocorrect rewrites the file to invalid Ruby:

```ruby
puts "x"y"
```

### Repro 2

Before autocorrect, the program prints:

```text
x#{1}y
```

After autocorrect, it becomes:

```ruby
puts "x1y"
```

and then prints:

```text
x1y
```

This looks related to how the cop rewrites string literal contents directly into the surrounding interpolated string context without preserving escaping requirements for the outer string.

## RuboCop configuration

```yaml
# No special configuration
```

## RuboCop version

```shell
1.87.0 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 4.0.2) +Rubydex [x86_64-linux]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.10.1
```

**Comment by RedZapdos123** [comment]
Opened a PR to resolve this issue.
